### PR TITLE
Docs/json2csv

### PR DESCRIPTION
**Pull request by wizzy25**
#### Summary

Similar to other similar PRs, this PR adds documentation for the `product-json-2-csv` package. 
Also related to #325 

**This PR does not merge to master. A final PR will be opened when this is done to address that!**

### CODE REVIEW

File: docs/cli/product-json-to-csv.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+##### Example
+```
+$ @commercetools@commercetools/product-json-to-csv --projectKey <project_key> --input <path_to_JSON_file> --template <path_to_template_file> --output <path_to_output_file>
+```
+
+#### CSV Parser Template
+A parser template defines the content of the resulting parsed CSV file, by listing wanted product attribute names as header row. The header column values will be parsed and the resulting CSV file will contain corresponding attribute values of the exported products.
+```
+# only productType.name, the variant id and localized name (english) will be exported
```
Comment (tdeekens) [inline docs/cli/product-json-to-csv.md]: Is this correctly formatted?

File: docs/cli/product-json-to-csv.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+### JS
+For more direct usage, it is possible to use this module directly
+
+#### Configuration
+
+The constructor accepts four arguments:
+- `apiConfig` (Object): `AuthMiddleware` options for authentication on the commercetools platform. (Required. See [here](https://commercetools.github.io/nodejs/sdk/api/sdkMiddlewareAuth.html#named-arguments-options))
+- `parserConfig` (Object): Internal Parse configurations
+  - `categoryBy` (String): Specify which identifier should be used to reference the categories (Options: `name`, `key`, `externalId` and `namedPath`. Default: `name`)
+  - `categoryOrderHintBy` (String): Specify which identifier should be used to reference the categoryOrderHints (Options: `name`, `key`, `externalId` and `namedPath`. Default: `name`)
```
Comment (tdeekens) [inline docs/cli/product-json-to-csv.md]: What does this mean? I think I just fail to follow 😢

File: docs/cli/product-exporter.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Product Exporter
 
-A package that helps with exporting [commercetools products](https://dev.commercetools.com/http-api-projects-products.html) from the [commercetools platform](http://dev.commercetools.com/).
+A package that helps with exporting [commercetools products](https://docs.commercetools.com/http-api-projects-products.html#product) from the [commercetools platform](http://dev.commercetools.com/).
```
Comment (PhilippSpo) [inline docs/cli/product-exporter.md]: 👍

File: docs/cli/product-json-to-csv.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
+# Product JSON to CSV Parser
+
+Convert [commercetools Products](http://dev.commercetools.com/http-api-projects-discountCodes.html) CSV data to JSON. See example below for CSV format, sample response and usage.
```
Comment (PhilippSpo) [inline docs/cli/product-json-to-csv.md]: Hmmmm isn't it the other way around? JSON data to CSV? I guess we can just remove this sentence in favor of the one below.

File: docs/cli/product-json-to-csv.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+A package that parses [commercetools products](https://docs.commercetools.com/http-api-projects-products.html#product) JSON data to CSV.
+The products to be parsed can either be read from a `.json` file or directly [piped in](http://www.gnu.org/software/bash/manual/bash.html#Pipelines) from the [product exporter](https://commercetools.github.io/nodejs/cli/product-exporter.html). exported in JSON format, or as chunks that can be piped to a parser for more export formats.
```
Comment (PhilippSpo) [inline docs/cli/product-json-to-csv.md]: I think there is a superfluous dot in this sentence right before "exported". Or the second sentence is missing its beginning 😄

File: docs/cli/product-json-to-csv.md (hunk 1 above)
Comment (PhilippSpo) [reply docs/cli/product-json-to-csv.md]: You mean it should be "product type name, ..."?

File: docs/cli/product-exporter.md (hunk 3 above)
Comment (adnasa) [inline docs/cli/product-exporter.md]: 👍 
On a side note: are all other `.md` updated to references the new URL?
We should change those, although not here

File: docs/cli/product-json-to-csv.md (hunk 4 above)
Comment (adnasa) [inline docs/cli/product-json-to-csv.md]: docs.commercetools.com?

File: docs/cli/product-json-to-csv.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
+# Product JSON to CSV Parser
+
+Convert [commercetools Products](http://dev.commercetools.com/http-api-projects-discountCodes.html) CSV data to JSON. See example below for CSV format, sample response and usage.
+
+A package that parses [commercetools products](https://docs.commercetools.com/http-api-projects-products.html#product) JSON data to CSV.
```
Comment (adnasa) [inline docs/cli/product-json-to-csv.md]: same here ☝️

File: docs/cli/product-json-to-csv.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+#### Configuration
+
+The constructor accepts four arguments:
+- `apiConfig` (Object): `AuthMiddleware` options for authentication on the commercetools platform. (Required. See [here](https://commercetools.github.io/nodejs/sdk/api/sdkMiddlewareAuth.html#named-arguments-options))
+- `parserConfig` (Object): Internal Parse configurations
+  - `categoryBy` (String): Specify which identifier should be used to reference the categories (Options: `name`, `key`, `externalId` and `namedPath`. Default: `name`)
+  - `categoryOrderHintBy` (String): Specify which identifier should be used to reference the categoryOrderHints (Options: `name`, `key`, `externalId` and `namedPath`. Default: `name`)
+  - `delimiter` (String): Delimiter used to separate cells in the output file (Default: `;`)
+  - `fillAllRows` (Boolean): Specify if product attributes like name should be added to each variant row (Default: `false`)
+  - `headerFields` (Array<String>): An array of header fields to be parsed to CSV (Optional. If omitted, a `.zip` file containing one csv file per product type will be created)
```
Comment (adnasa) [inline docs/cli/product-json-to-csv.md]: I don't understand this. care to add context?

File: docs/cli/product-json-to-csv.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+  - `language` (String): Default language used in resolving localised attributes (except lenums) and category names (Default: `en`)
```
Comment (adnasa) [inline docs/cli/product-json-to-csv.md]: Does this resolve the searchKeywords too on the product?

File: docs/cli/product-json-to-csv.md (hunk 1 above)
Comment (wizzy25) [reply docs/cli/product-json-to-csv.md]: It is. The `productType` field resolves the `productType.name`

File: docs/cli/product-json-to-csv.md (hunk 7 above)
Comment (wizzy25) [reply docs/cli/product-json-to-csv.md]: This section is supposed to be more technical. A header/template file can be passed through the cli, but if the module is to be used directly, this should be passed as as array. You should know how if you don't want to use the cli

File: docs/cli/product-json-to-csv.md (hunk 8 above)
Comment (wizzy25) [reply docs/cli/product-json-to-csv.md]: Yes

Comment (wizzy25) [reply docs/cli/product-json-to-csv.md]: But `searchKeywords` are always exported as localized strings with all available language, so this `language` option does not apply to it

File: docs/cli/product-json-to-csv.md (hunk 7 above)
Comment (adnasa) [reply docs/cli/product-json-to-csv.md]: > This section is supposed to be more technical. 

as a user of your cli, if I don't understand the explanation of what a parameter does, what's the point of using it

Comment (wizzy25) [reply docs/cli/product-json-to-csv.md]: It is explained above, but I'll make it clearer here as well

Comment (adnasa) [reply docs/cli/product-json-to-csv.md]: thanks 😘

**Comment by wizzy25** [comment]
What do yo guys think now?
@adnasa @tdeekens @junajan @PhilippSpo

File: docs/cli/product-exporter.md (hunk 3 above)
Comment (wizzy25) [reply docs/cli/product-exporter.md]: It's already merged to master

File: docs/cli/product-json-to-csv.md [hunk 9]
```diff
@@ -100,7 +100,7 @@ The constructor accepts four arguments:
   - `categoryOrderHintBy` (String): Specify which identifier should be used to reference the categoryOrderHints (Options: `name`, `key`, `externalId` and `namedPath`. Default: `name`)
   - `delimiter` (String): Delimiter used to separate cells in the output file (Default: `;`)
   - `fillAllRows` (Boolean): Specify if product attributes like name should be added to each variant row (Default: `false`)
-  - `headerFields` (Array<String>): An array of header fields to be parsed to CSV (Optional. If omitted, a `.zip` file containing one csv file per product type will be created)
+  - `headerFields` (Array<String>): An array of header fields to be passed to CSV. This headerFields array should contain the required columns of the CSV file(Optional. If omitted, a `.zip` file containing one csv file per product type will be created. This is synonymous with the `--template` flag in the CLI)
```
Comment (adnasa) [inline docs/cli/product-json-to-csv.md]: clearer now

**Comment by wizzy25** [comment]
Thanks guys. I'm coming with a final monster PR very soon 😂

**Pull request merged**